sky/cli.py

```diff
@@ -1325,6 +1325,7 @@ def launch(
     In both cases, the commands are run under the task's workdir (if specified)
     and they undergo job queue scheduling.
     """
+    backend_utils.check_network_connection()
```

Comment:
We also need to catch failures when running `sky exec` and also then invoking through the python API. I'm wondering if it is better to put this check lower in the stack, perhaps in `def _execute()` in execution.py?